src/merge.rs

```diff
@@ -1,11 +1,30 @@
-use libc::c_uint;
-use std::marker;
-use std::mem;
-use std::str;
-
-use crate::call::Convert;
-use crate::util::Binding;
-use crate::{raw, Commit, FileFavor, Oid};
+use libc::{c_int, c_uint, c_void};
+use std::{marker, mem, str};
+
+use crate::{call::Convert, panic, raw, util::Binding, Commit, FileFavor, Oid};
+
+pub type MergeheadForeachCb<'a> = dyn FnMut(&Oid) -> bool + 'a;
+
+pub struct MergeheadForeachCbData<'a> {
+    pub callback: &'a mut MergeheadForeachCb<'a>,
+}
+
+pub extern "C" fn mergehead_foreach_cb(oid: *const raw::git_oid, payload: *mut c_void) -> c_int {
```

Comment:
Could these definitions be located in `repo.rs`  since they're only used from there?

Reply:
sure I hesitated to do that before because right now nothing like this was in `repo.rs` but this gives us the added benefit of being able to mark a bunch more stuff non-`pub` 👍